### PR TITLE
create responsive image component

### PR DESCRIPTION
**Pull request by rizkybaihaqy**
## Type of changes

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Enhancement
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behaviour?

right now if we want a responsive image using next/image we need to wrap it in a div. if it only one image per page this is not a problem but if we have ten or more image which we want it to be responsive the source will be cluttered with too much image inside a div.

## What is the new behavior?

This MR is supposed to solve the issue I have addressed above. now we have Responsive Image component which accept src and alt as usual and accept className which we can define in the parent component.

## Checklist

- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have filled the `[Unreleased]` [Changelog](https://github.com/nodefluxio/weber/blob/main/CHANGELOG.md

## Security Pentester API Checklist
- [ ] IDOR
- [ ] Malicious Script Injection
- [ ] SQL Injection

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by williammfu — APPROVED** [review]
nice

File: frontend/app/components/modules/ResponsiveImage/ResponsiveImage.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+type Props = {
+  className?: string
+  src: string
+  alt: string
+}
+
+export const ResponsiveImage = ({ className, src, alt }: Props) => {
+  return (
+    <div className={className || styles.responsiveImage}>
+      <Image src={src} alt={alt} layout="fill" objectFit="cover" />
```
Comment (gregoriusjimmy) [inline frontend/app/components/modules/ResponsiveImage/ResponsiveImage.tsx:13]: gimana klo objectFitnya di kasih pilihan contain juga?

File: frontend/app/components/modules/ResponsiveImage/ResponsiveImage.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+type Props = {
+  className?: string
+  src: string
+  alt: string
+}
+
+export const ResponsiveImage = ({ className, src, alt }: Props) => {
+  return (
+    <div className={className || styles.responsiveImage}>
```
Comment (gregoriusjimmy) [inline frontend/app/components/modules/ResponsiveImage/ResponsiveImage.tsx:12]: containernya perlu di kasih style `position:relative` karena Image bakal jadi `position:absolute` dari Next.jsnya

Comment (williammfu) [reply frontend/app/components/modules/ResponsiveImage/ResponsiveImage.tsx:12]: class `responsiveImage` nya emang gaada ya?

File: frontend/app/components/modules/ResponsiveImage/ResponsiveImage.tsx (hunk 1 above)
Comment (rizkybaihaqy) [reply frontend/app/components/modules/ResponsiveImage/ResponsiveImage.tsx:13]: boleh si, tapi tanggung kyknya, gimana kalo gw kasih pilihan semua kemungkinan yang ada untuk [objectFit](https://developer.mozilla.org/en-US/docs/Web/CSS/object-fit)

File: frontend/app/components/modules/ResponsiveImage/ResponsiveImage.tsx (hunk 2 above)
Comment (rizkybaihaqy) [reply frontend/app/components/modules/ResponsiveImage/ResponsiveImage.tsx:12]: > containernya perlu di kasih style position:relative karena Image bakal jadi position:absolute dari Next.jsnya

ini gw buat kyk gini hanya sekedar fallback jika classNamenya ga dikasih. tapi bener juga harus dikasih position:absolute. oke ini gw ganti aja jadi selalu ada class ini dengan postion absolute.

> class responsiveImage nya emang gaada ya?

dan ini gw lupa isi.

File: frontend/app/components/modules/ResponsiveImage/ResponsiveImage.tsx (hunk 1 above)
Comment (rizkybaihaqy) [reply frontend/app/components/modules/ResponsiveImage/ResponsiveImage.tsx]: re-review on 14654f7

File: frontend/app/components/modules/ResponsiveImage/ResponsiveImage.tsx (hunk 2 above)
Comment (rizkybaihaqy) [reply frontend/app/components/modules/ResponsiveImage/ResponsiveImage.tsx]: re-review on dba75a6

**Review by gregoriusjimmy — APPROVED** [review]

**Pull request merged**